### PR TITLE
Scripts endpoint sorting by name

### PR DESCRIPTION
**Pull request by Levizar**
**Is your feature request related to a problem? Please describe.**
The paginated /scripts endpoint should be able to be sorted by Name

**Id**: 3c161d5

### CODE REVIEW

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/script/ScriptControllerTest.java [hunk 1]
```diff
@@ -313,6 +310,175 @@ void getAllPaginationTest() throws Exception {
 
     }
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/script/ScriptControllerTest.java:312]: Can you also create a test that contains all entries on 1page

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java [hunk 2]
```diff
@@ -259,20 +259,39 @@ private String getScriptAndScriptVRSTable(Pageable pageable, String scriptName,
...
     }
 
+    /**
+     * This method provide an Order by statement to order the ScriptAndScriptVersionTable
+     * @param pageable - pageable object containing or not the order object
+     * @return a String containing the ORDER BY statement
+     */
+    private String getOrderByStatementForScriptAndScriptVersionTable(Pageable pageable) {
+        if (pageable == null || !pageable.getSort().isSorted())
+            return " ";
```
[the remark this answers is not in the document]
Comment (Levizar) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java:280]: You are right, there is always a Pageable and it always contain a Sort item
I made a small test:
I created this: 
```Java    
    @GetMapping("/test")
    public String getTested(Pageable pageable) {
        System.out.println(pageable.toString());
        System.out.println(pageable.getSort().isSorted());
        return pageable.toString();
    }
```
I used GET http://localhost:8080/api/scripts/test

and the result is:
```
    Page request [number: 0, size 20, sort: UNSORTED]
    false
```

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/script/ScriptControllerTest.java (hunk 1 above)
Comment (Levizar) [reply core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/script/ScriptControllerTest.java:312]: These are Mocked data and thus will appear as I created them.
I will do it there and in the service to test the interaction with DB.

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Levizar) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java:280]: But I think it is better to keep it for several reason.

1. The IDE is complaining 🥇 
2. It is possible for the method to be used one day somewhere and having a null argument passed by. It is a safety to predict how the method should behave.
If the element is null -> it means you don't want a pagination -> you are given no pagination. 
Regarding the sorting: It has no point sorting in the SQL DB if you fetch everything.
At that point, if the server can handle such data, it is better to handle the sort on the server.

[the remark this answers is not in the document]
Comment (BerrevoetsRobbe) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java:301]: I don't get any IDE message if I make the change. Also, willingly providing a null parameter should not be encouraged. Should this ever be the case we can create a telescoped function using Pageable.unpaged()

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java [hunk 3]
```diff
@@ -196,6 +200,22 @@ private ScriptExecutionDto mapScriptExecutionDto(CachedRowSet cachedRowSet) thro
...
+        if (pageable == null || pageable.getSort().isUnsorted())
+            return;
+        pageable.getSort().forEach(order -> {
+            // add other sorting here
+            if (order.getProperty().equalsIgnoreCase("NAME")) {
+                scriptDtoList.sort((Comparator.comparing(ScriptDto::getName)));
+            }
+        });
+    }
+
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/dto/ScriptDtoRepository.java:218]: A more elagant way is to create a LinkedHashMap in the getAll() and getByName() function. Tests succeed if this change is made

**Pull request merged**